syntaxes/templ.tmLanguage.json

```diff
@@ -389,65 +386,63 @@
         }
       ]
     },
-    "empty-import-expression":{
-      "match": "@(?:[A-z_][A-z_0-9]*\\.)?[A-z_][A-z_0-9]*\\(.*\\)\\s*$",
-      "captures": {
-        "0": {
-          "patterns": [
-            {
-              "include": "#import-expression-start"
-            }
-          ]
-        }
-      }
-    },
     "import-expression": {
-      "begin": "@(?:[A-z_][A-z_0-9]*\\.)?[A-z_][A-z_0-9]*\\(.*\\)\\s*{$",
-      "beginCaptures": {
-        "0": {
+      "patterns": [
+        {
+          "name": "import-expression.templ",
+          "begin": "(@)((?:[A-z_][A-z_0-9]*\\.)?[A-z_][A-z_0-9]*\\()",
+          "beginCaptures": {
+            "1": {
+              "name": "keyword.control.go"
+            },
+            "2": {
+              "patterns": [
+                {
+                  "include": "source.go"
+                }
+              ]
+            }
+          },
+          "end": "\\s*$",
           "patterns": [
             {
-              "include": "#import-expression-start"
+              "name": "params.import-expression.templ",
+              "begin": "(?<=\\()",
+              "end": "(\\))",
+              "endCaptures": {
+                "1": {
+                  "name": "punctuation.definition.end.bracket.round.go"
+                }
+              },
+              "patterns": [
+                {
+                  "include": "source.go"
+                }
+              ]
+            },
+            {
+              "name": "children.import-expression.templ",
```

Comment:
Finally, this last inner pattern matches from the previous closing parenthesis and opening curly brace to the closing curly brace of the `templ` function. It provides the respective `punctuation.brace.*` to the curly braces and marks the captured children with the internal `#template-node` pattern.